manifests/charts/gateways/istio-egress/templates/deployment.yaml

```diff
@@ -131,7 +131,7 @@ spec:
             failureThreshold: 30
             httpGet:
               path: /healthz/ready
-              port: 15020
+              port: 15090
```

Comment:
I thought we decided not to change the health check port for the gateway?

Reply:
This move is orthogonal to that one but we should probably just do what we discussed in networking group immediately instead of doing this.

Really long winded thought process below ... see tl;dr at bottom

Currently we have a few endpoints:

* 15090 /stats -> envoy stats
* 15090 /healthz/ready -> 15020 /healthz/ready. The intent here is to ensure the readiness probe tests envoy connection as well rather than directly to pilot-agent. This is used for sidecar readiness probe, and with this PR gateway as well.
* 15020 /stats -> envoy + application + agent stats. This is disable on gateway for security
* 15020 /healthz/ready. This is used by AWS LB health check
* 15020 /... - > there are some other things on 15020 as well. These probably shouldn't be exposed externally to reduce attack surface but not a big issue

Ideally, we have two ports:
port0: health check. Exposed on public internet
port1: stats, app probers, quitquitquit, etc. Other internal stuff

However we probably want a way to hit envoy stats without going through the agent, so realistically this becomes 3 ports

port0: health check. Exposed on public internet
port1: stats merging, app probers, quitquitquit, etc. Other internal stuff to agent
port2: envoy stats

we also want to make sure we are backwards compatible with existing port #s

port0 does not need any compatibility; it only matters that it is "the first port"
port1: users call 15020/quitquitquit from their applications - we must preserve this port. Users **will** scrape prometheus on this port as well, but that has not yet shipped
port2: users scrape this port, we should preserve it

Therefor I suggest we map these to 150xx (new port), 15020, and 15090. This would result in:

150xx: Envoy listener on 0.0.0.0, forwarding /healthz/ready to 15020/healthz/ready
15090: Envoy listener on 0.0.0.0, forwarding to :15000/stats/prometheus (local envoy stats)
15020: pilot-agent listener. This does everything agent does today.

So I guess after all of this the tl;dr is we just move the health check to a brand new port? Any thoughts

Reply:
I've updated the PR to implement ^

Reply:
Is the primary issue that 15020 is used for both /quitquitquit and /healthz/ready today and we only want /healthz/ready to be exposed outside of the pod/mesh?